frontend/src/MuiForms/MetricItem.tsx

```diff
@@ -0,0 +1,51 @@
+import CloseIcon from '@mui/icons-material/Close'
+import { IconButton, Stack, TextField, Tooltip } from '@mui/material'
+import { useTheme } from '@mui/material/styles'
+import _ from 'lodash-es'
+import { useCallback, useState } from 'react'
+import MessageAlert from 'src/MessageAlert'
+import { MetricValueWithId } from 'src/MuiForms/Metrics'
+import { isValidNumber } from 'utils/stringUtils'
+
+interface MetricItemProps {
+  metric: MetricValueWithId
+  onChange: (updatedMetric: MetricValueWithId) => void
+  onDelete: (metricId: string) => void
+}
+
+export default function MetricItem({ metric, onChange, onDelete }: MetricItemProps) {
+  const theme = useTheme()
+  const [errorMessage, setErrorMessage] = useState('')
+
+  const handleMetricValueOnChange = useCallback(
+    (value: string) => {
+      const valueAsStringOrNumber: string | number = value
+      parseInt(valueAsStringOrNumber)
```

Comment:
I don't think this is useful until the value is validated. Also `parseInt` will convert an empty string to `NaN` which is invalid and makes means you have to delete the entry.

Reply:
I've removed it as it wasn't needed, but I didn't see it cause `Nan` issues when entering strings before weirdly.